### ISSUE TITLE
`ddev drush` commands hang indefinitely after a task was successfully finished

### ISSUE THREAD

**Issue by rpkoller** [drud/ddev#3708]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Run a Diagnostic and Paste Link Here

https://gist.github.com/rpkoller/43a21ef4ccbd6335657f5f003e3f1e98

### Current Behavior

Somewhere back in october or november 2021 an odd behavior using drush with ddev started. i am setting up sandbox drupal installations locally with ansible. during that process i am calling a few `ddev drush` commands. all of a sudden the process hung at one of the drush steps. the odd thing was that there was no ansible, ddev or drush update in a few weeks back then. so basically nothing changed. taking ansible out of the equation doing the steps manually (after thinking ansible was causing things first)  the hangs happened also for the stand alone `ddev drush` commands. the problem is the hangs dont happen consistently but sometimes it works sometimes it doesnt but then consistently for a while. therefor it was hard to debug and put into words what might causing it. 

i am running for example:

```
$> ddev drush cr
 [success] Cache rebuild complete.
```
but the prompt never shows up again no matter how long i wait when it happens. i have to force quit the command. but the point is the command successfully finishes. for example if i run `ddev en admin_toolbar`  and a hang takes place the module is properly enabled if i check after force quitting BUT without the force quit i would never get out of the `ddev en` process. 

i am using the standard drush command in `~/.ddev/commands/web/` - no modifications

```
#!/bin/bash

#ddev-generated
## Description: Run drush CLI inside the web container
## Usage: drush [flags] [args]
## Example: "ddev drush uli" or "ddev drush sql-cli" or "ddev drush --version"
## ProjectTypes: drupal7,drupal8,drupal9,drupal10,backdrop
## ExecRaw: true

if ! command -v drush >/dev/null; then
  echo "drush is not available. You may need to 'ddev composer require drush/drush'"
  exit 1
fi
drush "$@"
```

and running for example `ddev drush -vvv cr` isn't providing any further insights cuz there is no output after ` [success] Cache rebuild complete.` for `ddev drush cr`. it just hangs. 

the only thing i've noticed and figured. 

`ddev . drush cr` / `ddev exec drush cr`  works while after a computer startup the first run of for example `ddev drush cr` runs at least most of the time without a hang. first thought i would finally found a pattern but all of a sudden it also happened also on first run. but after that most of the time the hangs take place for `ddev drush` commands. 

@rfay mentioned in that context that `ddev mysql` uses "raw" in contrast to `ddev . mysql` if i remember correctly in the v.1.19.0-rc1 discussion thread on discord. so i figure it might be the similar case for `ddev drush` and `ddev . drush`. and that that differing throughput might cause that odd intermittent behaviour? 

### Expected Behavior

there shouldn't be any hangs after successfully running a drush command no matter if it is called via `ddev drush` or `ddev . drush`/`ddev exec drush`

**Comment by rfay** [comment]
Thanks for the issue!

I don't understand completely which command hangs (or if both do)? `ddev exec drush` and `ddev . drush` are the same thing. `ddev drush` is slightly different, as you correctly remember, because it's the equivalent of `ddev exec --raw drush`. 

Are you able to recreate this with `ddev exec mysql` or `ddev mysql` (you seem to have mentioned that command also.)

**Comment by rpkoller** [comment]
`ddev drush`is the one where i've noticed the hangs. `ddev . drush`and `ddev exec drush`i havent used as soon as i learned that `ddev drush` existed (was too lazy and preferred the `ddev drush` shorthand) or how to manually set it up back then (back when there was the example drush file you had to rename and enter the direct  path in the container). but  after i got remembered by you about the difference i started to test `ddev . drush`/`ddev exec drush` and it worked. with that i was unable to reproduce the hangs so far. stand alone commands as well as  altered ansible scripts all running through. just the shorthand `ddev drush`has that intermittent behaviour so far. 

never used  `ddev mysql` nor `ddev . mysql`. just picked that example up because you mentioned it on discord explaining the raw difference. the only direct command i use aside `ddev drush`is` ddev composer`. that works all the time no hangs. but i can try the mysql one later tonight when axe-con is over.

**Comment by rfay** [comment]
So you never see this with `ddev exec drush cr`, but you see it with `ddev drush cr`, correct? But it seems your example was with `ddev exec drush cr` ???

Can you make this happen with a plain vanilla D9 project? (Is it possible it could be something wrong with `drush cr` ?)

**Comment by ghalusa** [comment]
Confirmed, `ddev drush cr` hangs, while `ddev exec drush cr` works as it should. 

```
ddev version

 ITEM             VALUE
 DDEV version     v1.19.2
 architecture     amd64
 db               drud/ddev-dbserver-mariadb-10.3:v1.19.2
 dba              phpmyadmin:5
 ddev-ssh-agent   drud/ddev-ssh-agent:v1.19.0
 docker           20.10.14
 docker-compose   v2.3.3
 docker-platform  docker-desktop
 mutagen          0.12.0
 os               darwin
 router           drud/ddev-router:v1.19.0
 web              drud/ddev-webserver:v1.19.2
 ```

**Comment by rfay** [comment]
@ghalusa could you please do a `ddev debug test` and put the results in a gist on gist.github.com and provide the link here? Then please make sure you can replicate this on a plain vanilla [drupal 9 quickstart](https://ddev.readthedocs.io/en/latest/users/cli-usage/#drupal-9-quickstart)? If you could say something about your hardware and its vintage it might help too. Thanks!

**Comment by digitaldonkey** [comment]
Same issue. Same Versions. 
Docker "Clean / Purge data" and "Reset to factory defaults" helped finally. Puh. 

Maybe I updated ddev without `ddev poweroff` before to run into this.

**Comment by rfay** [comment]
Interesting, could be a docker desktop issue then. I assume you're on latest 4.8.1... and yes, upgrading over time on docker desktop can have some interesting side-effects, and reset to factory defaults could have affected something.

**Comment by rfay** [comment]
Unfortunately `ddev version` doesn't show your docker desktop version. You can get that with "about docker desktop"

**Comment by digitaldonkey** [comment]
```
Engine: 20.10.14
Compose: 1.29.2
Credential Helper: v0.6.4
Kubernetes: v1.24.0
Snyk: v1.827.0
```

**Comment by rfay** [comment]
Thanks, I didn't know they'd released 4.8.2. Oh more test runner updates yet again.

**Comment by rfay** [comment]
Haven't heard anything about this lately, so closing. Happy to continue the conversation here or elsewhere, and we can reopen this if it's reproducible.
